### ISSUE TITLE
fatal: Unable to mark file with untracked files in the repository

### ISSUE THREAD

**Issue by cod3gen** [ccxt/ccxt#4772]
- OS: Ubuntu 18.04 -> Docker
- CCXT version: ccxt@1.18.315
- Method: npm run build..

Hi @kroitor. Its been a couple weeks now since i have worked with ccxt, but after updating to latest available version, i am no longer able to compile my own custom exchange files by running npm run build, they are all ending with following error: 
- fatal: Unable to mark file python/ccxt/graviex3.py
- fatal: Unable to mark file python/ccxt/stex.py

At first i thought it may be some updates that have been happening on my local computer so i moved over to ubuntu with docker environment, and still getting the same error. Any ideas? Removing the two files and its builds as supposed too.

To bypass this i am currently just running transpile and then qa commands which does generate the files needed.

npm log;

> 0 info it worked if it ends with ok
> 1 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'run', 'build' ]
> 2 info using npm@6.7.0
> 3 info using node@v11.10.1
> 4 verbose run-script [ 'prebuild', 'build', 'postbuild' ]
> 5 info lifecycle ccxt@1.18.315~prebuild: ccxt@1.18.315
> 6 info lifecycle ccxt@1.18.315~build: ccxt@1.18.315
> 7 verbose lifecycle ccxt@1.18.315~build: unsafe-perm in lifecycle true
> 8 verbose lifecycle ccxt@1.18.315~build: PATH: /usr/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/ccxt/node_modules/.bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
> 9 verbose lifecycle ccxt@1.18.315~build: CWD: /ccxt
> 10 silly lifecycle ccxt@1.18.315~build: Args: [ '-c',
> 10 silly lifecycle   'npm run build-without-docs && npm run pandoc-all && npm run update-badges' ]
> 11 silly lifecycle ccxt@1.18.315~build: Returned: code: 1  signal: null
> 12 info lifecycle ccxt@1.18.315~build: Failed to exec build script
> 13 verbose stack Error: ccxt@1.18.315 build: `npm run build-without-docs && npm run pandoc-all && npm run update-badges`
> 13 verbose stack Exit status 1
> 13 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:301:16)
> 13 verbose stack     at EventEmitter.emit (events.js:197:13)
> 13 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
> 13 verbose stack     at ChildProcess.emit (events.js:197:13)
> 13 verbose stack     at maybeClose (internal/child_process.js:984:16)
> 13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:265:5)
> 14 verbose pkgid ccxt@1.18.315
> 15 verbose cwd /ccxt
> 16 verbose Linux 4.15.0-45-generic
> 17 verbose argv "/usr/bin/node" "/usr/bin/npm" "run" "build"
> 18 verbose node v11.10.1
> 19 verbose npm  v6.7.0
> 20 error code ELIFECYCLE
> 21 error errno 1
> 22 error ccxt@1.18.315 build: `npm run build-without-docs && npm run pandoc-all && npm run update-badges`
> 22 error Exit status 1
> 23 error Failed at the ccxt@1.18.315 build script.
> 23 error This is probably not a problem with npm. There is likely additional logging output above.
> 24 verbose exit [ 1, true ]

> Exchanges exported successfully. (<anonymous> @ export-exchanges.js:284)
> 
> > ccxt@1.18.315 git-ignore-generated-files /ccxt
> > node build/git-ignore-generated-files
> 
> Disabling the git changes tracking for 418 generated files... (<anonymous> @ git-ignore-generated-files.js:70)
> fatal: Unable to mark file python/ccxt/graviex3.py
> /ccxt/build/git-ignore-generated-files.js:88
>         if (!e.message.toLowerCase ().includes ('not a git repository')) throw e
>                                                                          ^
> 
> Error: Command failed: git update-index --assume-unchanged dist/ccxt.browser.js python/test/test_decimal_to_precision.py php/test/decimal_to_precision.php python/test/test_exchange_datetime_functions.py php/test/test_exchange_datetime_functions.php python/test/test.py doc/FAQ.rst doc/README.rst doc/exchanges-by-country.rst doc/exchanges.rst doc/install.rst doc/manual.rst python/README.rst python/ccxt/_1btcxe.py python/ccxt/async_support/_1btcxe.py php/_1btcxe.php python/ccxt/acx.py python/ccxt/async_support/acx.py php/acx.php python/ccxt/allcoin.py python/ccxt/async_support/allcoin.py php/allcoin.php python/ccxt/anxpro.py python/ccxt/async_support/anxpro.py php/anxpro.php python/ccxt/anybits.py python/ccxt/async_support/anybits.py php/anybits.php python/ccxt/bcex.py python/ccxt/async_support/bcex.py php/bcex.php python/ccxt/bibox.py python/ccxt/async_support/bibox.py php/bibox.php python/ccxt/bigone.py python/ccxt/async_support/bigone.py php/bigone.php python/ccxt/binance.py python/ccxt/async_support/binance.py php/binance.php python/ccxt/bit2c.py python/ccxt/async_support/bit2c.py php/bit2c.php python/ccxt/bitbank.py python/ccxt/async_support/bitbank.py php/bitbank.php python/ccxt/bitbay.py python/ccxt/async_support/bitbay.py php/bitbay.php python/ccxt/bitfinex.py python/ccxt/async_support/bitfinex.py php/bitfinex.php python/ccxt/bitfinex2.py python/ccxt/async_support/bitfinex2.py php/bitfinex2.php python/ccxt/bitflyer.py python/ccxt/async_support/bitflyer.py php/bitflyer.php python/ccxt/bitforex.py python/ccxt/async_support/bitforex.py php/bitforex.php python/ccxt/bithumb.py python/ccxt/async_support/bithumb.py php/bithumb.php python/ccxt/bitibu.py python/ccxt/async_support/bitibu.py php/bitibu.php python/ccxt/bitkk.py python/ccxt/async_support/bitkk.py php/bitkk.php python/ccxt/bitlish.py python/ccxt/async_support/bitlish.py php/bitlish.php python/ccxt/bitmarket.py python/ccxt/async_support/bitmarket.py php/bitmarket.php python/ccxt/bitmex.py python/ccxt/async_support/bitmex.py php/bitmex.php python/ccxt/bitsane.py python/ccxt/async_support/bitsane.py php/bitsane.php python/ccxt/bitso.py python/ccxt/async_support/bitso.py php/bitso.php python/ccxt/bitstamp.py python/ccxt/async_support/bitstamp.py php/bitstamp.php python/ccxt/bitstamp1.py python/ccxt/async_support/bitstamp1.py php/bitstamp1.php python/ccxt/bittrex.py python/ccxt/async_support/bittrex.py php/bittrex.php python/ccxt/bitz.py python/ccxt/async_support/bitz.py php/bitz.php python/ccxt/bl3p.py python/ccxt/async_support/bl3p.py php/bl3p.php python/ccxt/bleutrade.py python/ccxt/async_support/bleutrade.py php/bleutrade.php python/ccxt/braziliex.py python/ccxt/async_support/braziliex.py php/braziliex.php python/ccxt/btcalpha.py python/ccxt/async_support/btcalpha.py php/btcalpha.php python/ccxt/btcbox.py python/ccxt/async_support/btcbox.py php/btcbox.php python/ccxt/btcchina.py python/ccxt/async_support/btcchina.py php/btcchina.php python/ccxt/btcexchange.py python/ccxt/async_support/btcexchange.py php/btcexchange.php python/ccxt/btcmarkets.py python/ccxt/async_support/btcmarkets.py php/btcmarkets.php python/ccxt/btctradeim.py python/ccxt/async_support/btctradeim.py php/btctradeim.php python/ccxt/btctradeua.py python/ccxt/async_support/btctradeua.py php/btctradeua.php python/ccxt/btcturk.py python/ccxt/async_support/btcturk.py php/btcturk.php python/ccxt/buda.py python/ccxt/async_support/buda.py php/buda.php python/ccxt/bxinth.py python/ccxt/async_support/bxinth.py php/bxinth.php python/ccxt/ccex.py python/ccxt/async_support/ccex.py php/ccex.php python/ccxt/cex.py python/ccxt/async_support/cex.py php/cex.php python/ccxt/chbtc.py python/ccxt/async_support/chbtc.py php/chbtc.php python/ccxt/chilebit.py python/ccxt/async_support/chilebit.py php/chilebit.php python/ccxt/cobinhood.py python/ccxt/async_support/cobinhood.py php/cobinhood.php python/ccxt/coinbase.py python/ccxt/async_support/coinbase.py php/coinbase.php python/ccxt/coinbaseprime.py python/ccxt/async_support/coinbaseprime.py php/coinbaseprime.php python/ccxt/coinbasepro.py python/ccxt/async_support/coinbasepro.py php/coinbasepro.php python/ccxt/coincheck.py python/ccxt/async_support/coincheck.py php/coincheck.php python/ccxt/coinegg.py python/ccxt/async_support/coinegg.py php/coinegg.php python/ccxt/coinex.py python/ccxt/async_support/coinex.py php/coinex.php python/ccxt/coinexchange.py python/ccxt/async_support/coinexchange.py php/coinexchange.php python/ccxt/coinfalcon.py python/ccxt/async_support/coinfalcon.py php/coinfalcon.php python/ccxt/coinfloor.py python/ccxt/async_support/coinfloor.py php/coinfloor.php python/ccxt/coingi.py python/ccxt/async_support/coingi.py php/coingi.php python/ccxt/coinmarketcap.py python/ccxt/async_support/coinmarketcap.py php/coinmarketcap.php python/ccxt/coinmate.py python/ccxt/async_support/coinmate.py php/coinmate.php python/ccxt/coinnest.py python/ccxt/async_support/coinnest.py php/coinnest.php python/ccxt/coinone.py python/ccxt/async_support/coinone.py php/coinone.php python/ccxt/coinspot.py python/ccxt/async_support/coinspot.py php/coinspot.php python/ccxt/cointiger.py python/ccxt/async_support/cointiger.py php/cointiger.php python/ccxt/coolcoin.py python/ccxt/async_support/coolcoin.py php/coolcoin.php python/ccxt/coss.py python/ccxt/async_support/coss.py php/coss.php python/ccxt/crex24.py python/ccxt/async_support/crex24.py php/crex24.php python/ccxt/crypton.py python/ccxt/async_support/crypton.py php/crypton.php python/ccxt/cryptopia.py python/ccxt/async_support/cryptopia.py php/cryptopia.php python/ccxt/deribit.py python/ccxt/async_support/deribit.py php/deribit.php python/ccxt/dsx.py python/ccxt/async_support/dsx.py php/dsx.php python/ccxt/ethfinex.py python/ccxt/async_support/ethfinex.py php/ethfinex.php python/ccxt/exmo.py python/ccxt/async_support/exmo.py php/exmo.php python/ccxt/exx.py python/ccxt/async_support/exx.py php/exx.php python/ccxt/fcoin.py python/ccxt/async_support/fcoin.py php/fcoin.php python/ccxt/flowbtc.py python/ccxt/async_support/flowbtc.py php/flowbtc.php python/ccxt/foxbit.py python/ccxt/async_support/foxbit.py php/foxbit.php python/ccxt/fybse.py python/ccxt/async_support/fybse.py php/fybse.php python/ccxt/fybsg.py python/ccxt/async_support/fybsg.py php/fybsg.php python/ccxt/gatecoin.py python/ccxt/async_support/gatecoin.py php/gatecoin.php python/ccxt/gateio.py python/ccxt/async_support/gateio.py php/gateio.php python/ccxt/gdax.py python/ccxt/async_support/gdax.py php/gdax.php python/ccxt/gemini.py python/ccxt/async_support/gemini.py php/gemini.php python/ccxt/getbtc.py python/ccxt/async_support/getbtc.py php/getbtc.php python/ccxt/graviex3.py python/ccxt/async_support/graviex3.py php/graviex3.php python/ccxt/hadax.py python/ccxt/async_support/hadax.py php/hadax.php python/ccxt/hitbtc.py python/ccxt/async_support/hitbtc.py php/hitbtc.php python/ccxt/hitbtc2.py python/ccxt/async_support/hitbtc2.py php/hitbtc2.php python/ccxt/huobipro.py python/ccxt/async_support/huobipro.py php/huobipro.php python/ccxt/huobiru.py python/ccxt/async_support/huobiru.py php/huobiru.php python/ccxt/ice3x.py python/ccxt/async_support/ice3x.py php/ice3x.php python/ccxt/independentreserve.py python/ccxt/async_support/independentreserve.py php/independentreserve.php python/ccxt/indodax.py python/ccxt/async_support/indodax.py php/indodax.php python/ccxt/itbit.py python/ccxt/async_support/itbit.py php/itbit.php python/ccxt/jubi.py python/ccxt/async_support/jubi.py php/jubi.php python/ccxt/kkex.py python/ccxt/async_support/kkex.py php/kkex.php python/ccxt/kraken.py python/ccxt/async_support/kraken.py php/kraken.php python/ccxt/kucoin.py python/ccxt/async_support/kucoin.py php/kucoin.php python/ccxt/kucoin2.py python/ccxt/async_support/kucoin2.py php/kucoin2.php python/ccxt/kuna.py python/ccxt/async_support/kuna.py php/kuna.php python/ccxt/lakebtc.py python/ccxt/async_support/lakebtc.py php/lakebtc.php python/ccxt/lbank.py python/ccxt/async_support/lbank.py php/lbank.php python/ccxt/liqui.py python/ccxt/async_support/liqui.py php/liqui.php python/ccxt/liquid.py python/ccxt/async_support/liquid.py php/liquid.php python/ccxt/livecoin.py python/ccxt/async_support/livecoin.py php/livecoin.php python/ccxt/luno.py python/ccxt/async_support/luno.py php/luno.php python/ccxt/lykke.py python/ccxt/async_support/lykke.py php/lykke.php python/ccxt/mercado.py python/ccxt/async_support/mercado.py php/mercado.php python/ccxt/mixcoins.py python/ccxt/async_support/mixcoins.py php/mixcoins.php python/ccxt/negociecoins.py python/ccxt/async_support/negociecoins.py php/negociecoins.php python/ccxt/nova.py python/ccxt/async_support/nova.py php/nova.php python/ccxt/okcoincny.py python/ccxt/async_support/okcoincny.py php/okcoincny.php python/ccxt/okcoinusd.py python/ccxt/async_support/okcoinusd.py php/okcoinusd.php python/ccxt/okex.py python/ccxt/async_support/okex.py php/okex.php python/ccxt/paymium.py python/ccxt/async_support/paymium.py php/paymium.php python/ccxt/poloniex.py python/ccxt/async_support/poloniex.py php/poloniex.php python/ccxt/quadrigacx.py python/ccxt/async_support/quadrigacx.py php/quadrigacx.php python/ccxt/rightbtc.py python/ccxt/async_support/rightbtc.py php/rightbtc.php python/ccxt/southxchange.py python/ccxt/async_support/southxchange.py php/southxchange.php python/ccxt/stex.py python/ccxt/async_support/stex.py php/stex.php python/ccxt/stronghold.py python/ccxt/async_support/stronghold.py php/stronghold.php python/ccxt/surbitcoin.py python/ccxt/async_support/surbitcoin.py php/surbitcoin.php python/ccxt/theocean.py python/ccxt/async_support/theocean.py php/theocean.php python/ccxt/therock.py python/ccxt/async_support/therock.py php/therock.php python/ccxt/tidebit.py python/ccxt/async_support/tidebit.py php/tidebit.php python/ccxt/tidex.py python/ccxt/async_support/tidex.py php/tidex.php python/ccxt/uex.py python/ccxt/async_support/uex.py php/uex.php python/ccxt/upbit.py python/ccxt/async_support/upbit.py php/upbit.php python/ccxt/urdubit.py python/ccxt/async_support/urdubit.py php/urdubit.php python/ccxt/vaultoro.py python/ccxt/async_support/vaultoro.py php/vaultoro.php python/ccxt/vbtc.py python/ccxt/async_support/vbtc.py php/vbtc.php python/ccxt/virwox.py python/ccxt/async_support/virwox.py php/virwox.php python/ccxt/xbtce.py python/ccxt/async_support/xbtce.py php/xbtce.php python/ccxt/yobit.py python/ccxt/async_support/yobit.py php/yobit.php python/ccxt/yunbi.py python/ccxt/async_support/yunbi.py php/yunbi.php python/ccxt/zaif.py python/ccxt/async_support/zaif.py php/zaif.php python/ccxt/zb.py python/ccxt/async_support/zb.py php/zb.php
> fatal: Unable to mark file python/ccxt/graviex3.py
> 
>     at checkExecSyncError (child_process.js:637:11)
>     at execSync (child_process.js:674:13)
>     at gitUpdateIndex (/ccxt/build/git-ignore-generated-files.js:82:13)
>     at Object.<anonymous> (/ccxt/build/git-ignore-generated-files.js:71:5)
>     at Module._compile (internal/modules/cjs/loader.js:738:30)
>     at Object.Module._extensions..js (internal/modules/cjs/loader.js:749:10)
>     at Module.load (internal/modules/cjs/loader.js:630:32)
>     at tryModuleLoad (internal/modules/cjs/loader.js:570:12)
>     at Function.Module._load (internal/modules/cjs/loader.js:562:3)
>     at Function.Module.runMain (internal/modules/cjs/loader.js:801:12)
> npm ERR! code ELIFECYCLE
> npm ERR! errno 1
> npm ERR! ccxt@1.18.315 git-ignore-generated-files: `node build/git-ignore-generated-files`
> npm ERR! Exit status 1
> npm ERR!
> npm ERR! Failed at the ccxt@1.18.315 git-ignore-generated-files script.
> npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     /root/.npm/_logs/2019-03-02T18_35_44_532Z-debug.log
> npm ERR! code ELIFECYCLE
> npm ERR! errno 1
> npm ERR! ccxt@1.18.315 build-without-docs: `npm run export-exchanges && npm run git-ignore-generated-files && npm run vss && npm run check-js-syntax && npm run browserify && npm run transpile && npm run qa`
> npm ERR! Exit status 1
> npm ERR!
> npm ERR! Failed at the ccxt@1.18.315 build-without-docs script.
> npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     /root/.npm/_logs/2019-03-02T18_35_44_551Z-debug.log
> npm ERR! code ELIFECYCLE
> npm ERR! errno 1
> npm ERR! ccxt@1.18.315 build: `npm run build-without-docs && npm run pandoc-all && npm run update-badges`
> npm ERR! Exit status 1
> npm ERR!
> npm ERR! Failed at the ccxt@1.18.315 build script.
> npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     /root/.npm/_logs/2019-03-02T18_35_44_565Z-debug.log

**Comment by cod3gen** [comment]
@kroitor Not only in Docker enviroment. Same on my local computer which is run without docker

**Comment by kroitor** [comment]
You should remove those two files from there and then run `npm run build`. 

These aren't merged into the main branch yet, which is causing the conflict on your side:
```
fatal: Unable to mark file python/ccxt/graviex3.py
fatal: Unable to mark file python/ccxt/stex.py
```

After building in a clean repo, put those two files there and run `npm run transpile` and `npm run export-exchanges`. Let us know if that doesn't solve the issue.

**Comment by xpl** [comment]
@cod3gen It seems that the issue is still there, I'm fixing it.

**Comment by xpl** [comment]
@cod3gen Should be fixed by ef8d24098fbab6499706a16b21506af6a258876f, now untracked files are explicitly ignored from that build step that had been failing.

Also I've added a softer error handling there so that even if any unexpected error occur in the future in that build step, it will allow the `npm run build` to continue, as that build step is not critical for building the project.

Let me know if you're still experiencing the issue with `npm run build`.
